### PR TITLE
docs: add section on using the driver with plain RDS databases

### PR DESCRIPTION
**Pull request by crystall-bitquill**
### Summary

Add section on using the driver with plain RDS databases

### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/using-the-jdbc-driver/UsingTheJdbcDriver.md [hunk 1]
```diff
@@ -101,20 +104,20 @@ DriverConfigurationProfiles.addOrReplaceProfile(
...
+| Data Cache Connection Plugin                                                                   | `dataCache`               | Plain RDS, Aurora      | Caches results from SQL queries matching the regular expression specified in the  `dataCacheTriggerCondition` configuration parameter.                                                                                                                                                                                                                                                                                                                                                                                 |
```
Comment (sergiyvamz) [inline docs/using-the-jdbc-driver/UsingTheJdbcDriver.md:112]: Any database?

File: docs/using-the-jdbc-driver/UsingTheJdbcDriver.md [hunk 2]
```diff
@@ -101,20 +104,20 @@ DriverConfigurationProfiles.addOrReplaceProfile(
...
+| Execution Time Connection Plugin                                                               | `executionTime`           | Plain RDS, Aurora      | Logs the time taken to execute any JDBC method.                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```
Comment (sergiyvamz) [inline docs/using-the-jdbc-driver/UsingTheJdbcDriver.md:113]: Any database?

File: docs/using-the-jdbc-driver/UsingTheJdbcDriver.md [hunk 3]
```diff
@@ -101,20 +104,20 @@ DriverConfigurationProfiles.addOrReplaceProfile(
...
+| Log Query Connection Plugin                                                                    | `logQuery`                | Plain RDS, Aurora      | Tracks and logs the SQL statements to be executed. Sometimes SQL statements are not passed directly to the JDBC method as a parameter, such as [executeBatch()](https://docs.oracle.com/javase/8/docs/api/java/sql/Statement.html#executeBatch--). Users can set `enhancedLogQueryEnabled` to `true`, allowing the JDBC Wrapper to obtain SQL statements via Java Reflection. <br><br> :warning:**Note:** Enabling Java Reflection may cause a performance degradation.                                                |
```
Comment (sergiyvamz) [inline docs/using-the-jdbc-driver/UsingTheJdbcDriver.md:114]: Any database?

File: docs/using-the-jdbc-driver/UsingTheJdbcDriver.md [hunk 4]
```diff
@@ -101,20 +104,20 @@ DriverConfigurationProfiles.addOrReplaceProfile(
...
+| [Driver Metadata Connection Plugin](./using-plugins/UsingTheDriverMetadataConnectionPlugin.md) | `driverMetaData`          | Plain RDS, Aurora      | Allows user application to override the return value of `DatabaseMetaData#getDriverName`                                                                                                                                                                                                                                                                                                                                                                                                                               |
```
Comment (sergiyvamz) [inline docs/using-the-jdbc-driver/UsingTheJdbcDriver.md:119]: Any database?

**Review by karenc-bq — APPROVED** [review]

**Pull request merged**